oscar_odin/mappings/_model_mapper.py

```diff
@@ -18,21 +20,28 @@ def __new__(cls, name, bases, attrs):
         mapping_type.one_to_one_fields = one_to_one_fields = []
         mapping_type.many_to_one_fields = many_to_one_fields = []
         mapping_type.many_to_many_fields = many_to_many_fields = []
-        related_object = getmeta(mapping_type.to_obj).related_objects
-        for relation in related_object:
+        mapping_type.foreign_key_fields = foreign_key_fields = []
+
+        meta = getmeta(mapping_type.to_obj)
+
+        for relation in meta.related_objects:
             if relation.many_to_many:
-                one_to_one_fields.append(
-                    (relation.related_name, relation.related_model._meta.db_table)
+                many_to_many_fields.append(
+                    (relation.related_name, relation.field.attname)
                 )
             elif relation.many_to_one:
                 many_to_one_fields.append(
-                    (relation.related_name, relation.related_model._meta.db_table)
+                    (relation.related_name, relation.field.attname)
                 )
             elif relation.one_to_many:
                 many_to_many_fields.append(
-                    (relation.related_name, relation.related_model._meta.db_table)
+                    (relation.related_name, relation.field.attname)
                 )
 
+        for field in meta.fields:
+            if isinstance(field, ForeignKey):
+                foreign_key_fields.append(("%s_id" % field.name, field.attname))
```

Comment:
field.attname is typically f"{field.name}_id". I would just use that. 

This might not even be required as the product is simply assigned to the parent instance anyway.